engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -249,15 +263,15 @@ object PartialWorkflowSources {
     else convertStringToUrl(workflowUrl)
   }
 
-  private def toMap(someInput: Option[String]): Map[String, JsValue] = {
-    import spray.json._
+  private def toMap(someInput: Option[String]): ErrorOr[Map[String, JsValue]] = {
     someInput match {
-      case Some(inputs: String) => inputs.parseJson match {
-        case JsObject(inputMap) => inputMap
-        case _ =>
-          throw new RuntimeException(s"Submitted inputs couldn't be processed, please check for syntactical errors")
+      case Some(input: String) =>  {
+        Try(input.parseJson).toErrorOrWithContext(s"parse input: '$input', which is not a valid json. Please check for syntactical errors.") flatMap {
+          case JsObject(inputMap) => inputMap.validNel
+          case j: JsValue => s"Submitted input '$input' belongs to ${j.getClass}, which is not a valid json object.".invalidNel
```

Comment:
ToL: Maybe something more along the lines of s"Submitted input '$input' of type ${j.getClass.getSimpleName} is not a JSON object"

Reply:
👍